src/Gaffer/Animation.cpp

```diff
@@ -302,16 +371,16 @@ const Animation::Key *Animation::CurvePlug::closestKey( float time ) const
 	Keys::const_iterator rightIt = m_keys.lower_bound( time );
 	if( rightIt == m_keys.end() )
 	{
-		return m_keys.rbegin()->get();
+		return &( *( m_keys.rbegin() ) );
 	}
-	else if( (*rightIt)->getTime() == time || rightIt == m_keys.begin() )
+	else if( (*rightIt).m_time == time || rightIt == m_keys.begin() )
```

Comment:
There are quite a few spots where we've kept the previous awkward combo of `(*it)->foo` (but transforming it to `(*it).foo`). Since we're storing `Key` rather than `KeyPtr` now, this could be a bit more readable as the more idiomatic `rightIt->m_time`. Might be worth cleaning up for readability's sake?

Reply:
I prefer ```(*it).foo``` as it describes whats actually happening, we are dereferencing an iterator and using the resulting value_type as a reference.

Reply:
Is that just for iterators, or for pointers too?